### ISSUE TITLE
vert, face switched in surface loading for geodesic distance calculations?

### ISSUE THREAD

**Issue by MrBurnst** [murraylab/brainsmash#23]
Hi!

I'm having trouble generating (dense) geodesic distance matrices from my surfaces. I have .surf.gii files (they are fine, as by loading them into connectome workbench), but when I try using them to generate geodesic distance matrices per

https://github.com/murraylab/brainsmash/blob/2087867d2e64cb41b61060e01f6de42675c42229/brainsmash/workbench/geo.py#L103

(and also using "use_wb") the vertices are loaded as faces and vice-versa. Eg. I'm getting #faces x #faces matrices instead of #vertices x #vertices. I checked and I think "vert" and "faces" are switched in the loading procedure:

```
>>> vert, faces = [darray.data for darray in nib.load(surface).darrays]  
>>> vert
array([[    0, 10244, 10242],
       [    0, 10242, 10245],
       [    0, 10245, 10247],
       ...,
       [40961, 39990,  9918],
       [40719, 39990, 40961],
       [40719,    11, 39990]], dtype=int32)
>>> faces
array([[-0.46004283,  0.04199572,  0.42387655],
       [-0.2947536 , -0.32666323,  0.49247652],
       [-0.04152068,  0.16850185,  0.40763494],
       ...,
       [-0.40267348, -0.44032592, -0.31085908],
       [-0.40869153, -0.4656167 , -0.30369812],
       [-0.41492808, -0.4888297 , -0.2946863 ]], dtype=float32)
```

Is that possible? I tried with all sorts of versions of nibabel, installing brainsmash via pip and github, always the same.

cheers,
Ernst

**Comment by MrBurnst** [comment]
actually, I think I figured out the issue (at least for the wb_command-based geodesic computation): some temporary .func.gii files are being written during loading, but they only contain the first dimension of the original data (eg. the face list). I changed https://github.com/murraylab/brainsmash/blob/2087867d2e64cb41b61060e01f6de42675c42229/brainsmash/utils/dataio.py#L164
to 
`return nib.load(filename).darrays[-1].data`

which seems to work.

**Comment by jbburt** [comment]
Hey Ernst, thanks for your question. If I may ask, where did you get your surface file? I just checked several atlases (eg HCP, Conte69, F99) and every surface file I tried to open had vertices before faces, ie vertexes were at index 0 in `darrays`... Maybe the real issue here is that the order is not standardized and the two can appear in reversed order? Assuming this is the case, I'm still a bit confused:

1. You say that you're using "use_wb" but [the line that you mentioned in your first comment](https://github.com/murraylab/brainsmash/blob/2087867d2e64cb41b61060e01f6de42675c42229/brainsmash/workbench/geo.py#L103) is only ever reached if you're **not** using use_wb. 

2. I'm also confused about how the change you made in your second comment resolved the issue... when `wb_command -surface-geodesic-distance` is called from the command line, the resulting `.func.gii` file has only a single element in `darrays` (at least this is the case when I try it). In other words, when opening the `.func.gii` file, `nib.load(filename).darrays[-1].data` is equivalent to `nib.load(filename).darrays[0].data`.

**Comment by MrBurnst** [comment]
Hi! thanks for looking into this. So, you are absolutely right, the issue was with my surface files. I generate them in matlab and store them using the HCP cifti tools, but I might be using a super-old version or something.
Anyways, the reason why my change works is that my surfaces have their faces stored first (maybe the data fields are ordered alphabetically? just guessing) - then, the load() function called by cortex(...) first loads the surface file and determines the number of vertices from the _last_ dimension (which are the vertices, in my case). Then, using `use_wb=True`, `load` is used for loading the output of wb_command, which only has one dimension, so `[-1] == [0]`. `use_wb=False` will not work bcs. of how `vert, faces` are populated in geo.py.
Long story short, switching dimensions works for me using `use_wb` - that's fine for now, and since I seem to be the only person who had this issue, it's probably the fault of some old Matlab tool I'm using.
